ext/opcache/zend_shared_alloc.c

```diff
@@ -594,6 +594,22 @@ void zend_accel_shared_protect(int mode)
 	for (i = 0; i < ZSMMG(shared_segments_count); i++) {
 		mprotect(ZSMMG(shared_segments)[i]->p, ZSMMG(shared_segments)[i]->size, mode);
 	}
+#elif defined(ZEND_WIN32)
+	int i;
+
+	if (!smm_shared_globals) {
+		return;
+	}
+
+	if (mode) {
+		mode = PAGE_READONLY;
+	} else {
+		mode = PAGE_READWRITE;
+	}
+
+	for (i = 0; i < ZSMMG(shared_segments_count); i++) {
+		VirtualProtect(ZSMMG(shared_segments)[i]->p, ZSMMG(shared_segments)[i]->size, mode, NULL);
```

Comment:
This doesn't look right according to the documentation:

`lpflOldProtect`
A pointer to a variable that receives the previous access protection value of the first page in the specified region of pages. If this parameter is NULL or does not point to a valid variable, the function fails.

Nor do we check the actual return value of `VirtualProtect`, unless it is intended to ignore a potential error, as `mprotect` also skips the return value checking